libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1014,7 +1014,7 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 						fatalTypeError(components[i]->location(), "Invalid mobile type.");
 
 					if (i == 0)
-						inlineArrayType = types[i];
+						inlineArrayType = types[i]->mobileType();
 					else if (inlineArrayType)
 						inlineArrayType = Type::commonType(inlineArrayType, types[i]);
```

Comment:
Does it make sense to take `types[i]->mobileType()` here as well, for the sake of symmetry?